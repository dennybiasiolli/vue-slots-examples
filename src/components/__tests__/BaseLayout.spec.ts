import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import BaseLayout from '../BaseLayout.vue'

describe('BaseLayout', () => {
  it('with default content', () => {
    const wrapper = mount(BaseLayout)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('with all slots', () => {
    const wrapper = mount(BaseLayout, {
      slots: {
        header: 'header content',
        default: 'Foo',
        footer: 'footer content'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
