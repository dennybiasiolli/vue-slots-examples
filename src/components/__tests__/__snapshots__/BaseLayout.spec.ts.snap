// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseLayout > with all slots 1`] = `
"<div class=\\"container\\">
  <header>header content</header>
  <main>Foo</main>
  <footer>footer content</footer>
</div>"
`;

exports[`BaseLayout > with default content 1`] = `
"<div class=\\"container\\">
  <header></header>
  <main></main>
  <footer></footer>
</div>"
`;
